book/whats_new/aug-18/swift-5.0-migration.md

```diff
@@ -0,0 +1,88 @@
+    * Use the new cocoapods CDN source:
+        * Remove `source 'git@github.com:CocoaPods/Specs.git'`.
+        * Add the new CDN `source  'https://cdn.cocoapods.org/'` (If you have private specs, add the new source below all of the other sources).
+        * Run `pod update`.
+8. Open the workspace with Xcode 10.2.1.
+9. For each one of the product targets make sure there isn't an override swift version (search for `Swift Language Version` inside each of the targets build settings and make sure the text is not in bold. They should show Swift 5`
+10. Build the framework in Xcode. Make sure the framework compiles with no warning or errors generated by the framework itself. Dependencies might create warnings that are out of our control.
+11. Increment the plist framework version to a major change version notion. For the record, Applicaster's versioning notion is `Major.Minor.Patch`.
+12. Push your code with GIT and make sure circle-ci build process ended successfully.
+13. Verify that a new podspec file was created inside Applicaster's private [CocoaPods-Private](https://github.com/applicaster/CocoaPods-Private) or public [CocoaPods](https://github.com/applicaster/CocoaPods) repositories.
+14. Create updated manifest before pushing the code.
+15. Upload after successfully passing circle ci
+
+## 3rd Party Plugins Migration
+
+The following instructions are for 3th party developers that are not using the applicaster closed frameworks logic.
+
+1. Install xcode 10.2.1 and cocoapods version 1.7.5 using the following command:
+    ```
+    sudo gem install cocoapods -v "=1.7.5"
+    ```
+2. Update podspec dependencies to support swift 5.0.
+    * Inside each podspec file, make sure that the s.xcconfig.'SWIFT_VERSION' value is set to 5.0.
+    * Update all dependencies to support swift 5.0 specs.
+3. Increment the podspec to a major change version notion in the specs directory. For the record, Applicaster's versioning notion is `Major.Minor.Patch`.
+4. Create an empty xcode test project (or update existing) with xocde 10.2.1 and a new podfile that points to your upgraded podspec file.
+5. Update podfile structure according to the following steps:
+    * dependencies to support swift 5.0 specs.
+    * Use the new cocoapods CDN source:
+        * Remove `source 'git@github.com:CocoaPods/Specs.git'`.
+        * Add the new CDN `source  'https://cdn.cocoapods.org/'` (If you have private specs, add the new source below all of the other sources).
+        * Run `pod update`.
+6. Update the podfile dependencies to support swift 5.0 specs.
+7. For each one of the product targets make sure there isn't an override swift version (search for `Swift Language Version` inside each of the targets build settings and make sure the text is not in bold. They should show Swift 5`
+8. Build the test project and make sure it’s working without errors.
+10. Create updated manifest before pushing the code.
+11. Push your code using git.
```

Comment:
better add a tag before pushing, and also add make sure CI passes and relevant deployment steps (if not built from source) succeed.

Reply:
On 3rd Party Plugins Migration, we don't any CI. 
It's under the responsibility of the 3rd developer.

Reply:
I'm doing the tag after the push to master state. 
The developer can decide here what to do before what.